dvc/ignore.py

```diff
@@ -72,36 +58,32 @@ def __call__(self, root, dirs, files):
 
 
 class DvcIgnoreFilter(object):
-    def __init__(self, wdir, ignore_file_handler=None):
+    def __init__(self, wdir, tree):
```

Comment:
Maybe it would make sense to just make it a part of the tree itself? dvc_walk and walk_files are also not used without the `tree`, so it seems to make sense to move those to Tree as well.

Reply:
But `dvc_walk()` is used in `get_mtime_and_size()`, do you propose in wrapping path into a tree there. I would leave `dvc_walk()` to do that. Ultimately we might not even need that tree.